docs/openapi/definitions/auth.yaml

```diff
@@ -0,0 +1,217 @@
+  version: 0.1.1-alpha.7
+externalDocs:
+  description: Backstage official documentation
+  url: https://github.com/spotify/backstage/blob/master/docs/README.md
+servers:
+  - url: http://localhost:7000/auth/google
+tags:
+  - name: provider
+    description: List of endpoints per provider
+paths:
+  /start:
+    get:
+      tags:
+        - provider
+      summary:
+        Initializes the authorization flow and redirects to a consent popup for
+        the provider
+      description:
+        The popup window is pointed to the `/start` endpoint of the auth
+        provider in the auth-backend plugin, which then redirects to the OAuth
+        consent screen of the provider.
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        302:
+          description: Redirect to the consent popup
+          headers:
+            Location:
+              description: The url to the consent popup
+              schema:
+                type: string
+            Set-Cookie:
+              description: Nonce cookie
+              schema:
+                $ref: '#/components/schemas/Nonce-Cookie'
+        default:
+          description: An error occurred
+  /frame/handler:
+    get:
+      tags:
+        - provider
+      summary: Handles the response from the consent popup for the provider
+      description:
+        If the login request is accepted, the popup window will be redirected
+        back to the `/handler/frame` endpoint of the auth backend.
+      parameters:
+        - name: scope
+          in: query
+          description: Scope requested for the auth session
+          required: true
+          schema:
+            type: string
+            example: email
+        - name: code
+          in: query
+          description:
+            Short-term authorization code used to be exchanged for long-term
+            tokens
+          required: true
+          schema:
+            type: string
+        - name: state
+          in: query
+          description: Nonce value stored in state
+          required: true
+          schema:
+            type: string
+        - name: provider-nonce
+          in: cookie
+          description: Nonce value stored in cookie
+          required: true
+          schema:
+            type: string
+        - name: env
+          in: query
+          description: Parameter to indicate runtime environment
+          required: true
+          schema:
+            type: string
+            example: development
+      responses:
+        200:
+          description: Message received from consent popup
```

Comment:
Should describe the postMessage logic here in more detail imo. Preferable at a lever where someone else would be able to implement it based on the spec, so that would include the different types of messages being passed.

Reply:
@Rugvip Take a look at the latest commit and let me know what you think?

Reply:
I'd probably assume that this was received in the response body, which isn't really true. The content definitely isn't application/json, so maybe it's still possible to switch to text/html, document how it works, and then use the schema to define the message body?